### PR TITLE
FIX Don't set up database etc if not needed

### PR DESCRIPTION
**Pull request by GuySartorelli**
Requires https://github.com/silverstripe/gha-generate-matrix/pull/77 to be merged and tagged first.

## Issue
- https://github.com/silverstripe/gha-ci/issues/97

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -664,7 +664,18 @@ jobs:
...
+          NEEDS_FULL_SETUP: ${{ matrix.needs_full_setup }}
         run: |
+          # Artifacts directory must be created after composer install as it would remove the artifacts directory
+          # This seems a bit strange, if you need to add an artifact at an earlier stage then revalidate that
+          # composer install does actually remove the artifact directory
+          mkdir artifacts
+
+          if [[ NEEDS_FULL_SETUP == "false" ]]; then
+            echo "skipping .env, database, and other full site setup steps"
+            exit 0
```
Comment (emteknetnz) [inline .github/workflows/ci.yml]: Instead of existing which may mess up the rest of the job, just move the block of code inside the conditional

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ChloeSartorelli) [reply .github/workflows/ci.yml]: Done

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -664,113 +664,118 @@ jobs:
...
-          # Note: the wonky indentation is intentional so there is no space at the start of
-          # each newline in the .env file
-          if [[ "${{ matrix.db }}" =~ mysql ]]; then
-            if [[ "${{ matrix.db }}" == "mysql57pdo" ]]; then
+          # Artifacts directory must be created after composer install as it would remove the artifacts directory
+          # This seems a bit strange, if you need to add an artifact at an earlier stage then revalidate that
+          # composer install does actually remove the artifact directory
+          mkdir artifacts
+
+          if [[ NEEDS_FULL_SETUP == "false" ]]; then
```
Comment (emteknetnz) [inline .github/workflows/ci.yml]:
```suggestion
          if [[ $NEEDS_FULL_SETUP == 'false' ]]; then
```
Need a `$` for the variable :-)

Using single quotes just to copy style above where a conditional check was done `if [[ $SUCCESSFULLY_REQUIRED == 'false' ]]; then`

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ChloeSartorelli) [reply .github/workflows/ci.yml]: The single/double quotes isn't consistent everywhere in this file (I copied double quotes from somewhere in here too) but I have no issues with that change.

Done.

**Review by emteknetnz — APPROVED** [review]

**Comment by emteknetnz** [comment]
@GuySartorelli Once this has auto-tagged could you manually re-trigger a recipe + a regular module + something without installer just to double check things still work as expected

**Pull request merged**